src/Cli/Microsoft.TemplateEngine.Cli/TelemetryHelper.cs

```diff
@@ -1,72 +1,38 @@
-        {
-            if (string.IsNullOrEmpty(paramName) || string.IsNullOrEmpty(inputParamValue))
-            {
-                return defaultValue;
-            }
-            ITemplateParameter? parameter = template.ParameterDefinitions.FirstOrDefault(x => string.Equals(x.Name, paramName, StringComparison.Ordinal));
-            if (parameter == null || parameter.Choices == null || parameter.Choices.Count == 0)
-            {
-                return defaultValue;
-            }
-            // This is a case-insensitive key lookup, because that is how Choices is initialized.
-            if (parameter.Choices.ContainsKey(inputParamValue))
-            {
-                return inputParamValue;
-            }
-            return defaultValue;
-        }
-
         /// <summary>
-        /// // The hashed mac address needs to be the same hashed value as produced by the other distinct sources given the same input. (e.g. VsCode).
+        ///  Checks if the <paramref name="parameterValues"/> contains a valid value for choice parameter name <paramref name="parameterName"/>.
+        ///  If so, returns hashed value of parameter value, otherwise <see langword="null"/>.
         /// </summary>
-        internal static string? Hash(string? text)
-        {
-            var sha256 = SHA256.Create();
-            return HashInFormat(sha256, text);
-        }
-
-        internal static string? HashWithNormalizedCasing(string? text)
-        {
-            if (text == null)
-            {
-                return null;
-            }
-
-            return Hash(text.ToUpper());
-        }
-
-        private static string? HashInFormat(SHA256 sha256, string? text)
+        internal static string? PrepareHashedChoiceValue(ITemplateInfo template, IReadOnlyDictionary<string, string?> parameterValues, string parameterName)
         {
-            if (text == null)
+            if (parameterValues.TryGetValue(parameterName, out string? value))
             {
-                return null;
-            }
+                if (value != null)
+                {
+                    ITemplateParameter? parameter = template.ParameterDefinitions.FirstOrDefault(x => string.Equals(x.Name, parameterName, StringComparison.Ordinal));
+                    if (parameter == null || parameter.Choices == null || parameter.Choices.Count == 0)
+                    {
+                        //not a choice parameter
+                        return null;
+                    }
 
-            byte[] bytes = Encoding.UTF8.GetBytes(text);
-            byte[] hash = sha256.ComputeHash(bytes);
-            StringBuilder hashString = new StringBuilder();
-            foreach (byte x in hash)
-            {
-                hashString.AppendFormat("{0:x2}", x);
+                    if (parameter.Choices.ContainsKey(value))
```

Comment:
I know this is currently being used only for 'single choices', but to prevent future missuse this should be either documented in xml comment (or method name) or multichoices should be handled as well (e.g. by first tokenizing by `TokenizeMultiValueParameter`, validating each value and then re-join them and hash the result - this will work for single choices as well, with same hash as the current code)

Reply:
Implemented it now, not to forget.